crates/datasources/Cargo.toml

```diff
@@ -59,6 +59,9 @@ dashmap = "5.5.0"
 calamine = { version = "0.22.1", features = ["dates"] }
 tiberius = { version = "0.12.2", default-features = false, features = ["tds73", "rustls", "chrono"] }
 lance = { git = "https://github.com/universalmind303/lance", rev = "81158eb540ff88ab5b4fce3a1170447760137412" }
+bson = "2.7.0"
+arrow-schema = "47.0.0"
+arrow-array = "47.0.0"
```

Comment:
We shouldn't be adding in arrow dependencies directly and instead should use the arrow crate that's re-exported by datafusion (e.g. `use datafusion::arrow::datatypes::SchemaRef;`)